src/Crocodealer/Command/Sync.hs

```diff
@@ -2,80 +2,62 @@ module Crocodealer.Command.Sync
+        repoFile <- fetchGitHubFile userName repo fileName
+        let fileDiff = calculateDiff templateFile repoFile
+        printFileDiff userName repo fileDiff
+
+fetchGitHubFile :: UserName -> RepoName -> FileName -> IO (Maybe ByteString)
+fetchGitHubFile userName repoName fileName = do
+    let url = "https://raw.githubusercontent.com/" <> unUserName userName <> "/" <> unRepoName repoName <> "/master/" <> unFileName fileName
     req <- parseRequest $ toString url
-    httpBS req
-
-printColors :: [Repo] -> IO ()
-printColors = mapM_ colorStatus
-  where
-    colorStatus :: Repo -> IO ()
-    colorStatus status =
-        case status of
-            Repo user repoName UpToDate ->
-                infoMessage $ user <> "/" <> repoName <> ": " <> "Up-to-date"
-            Repo user repoName Outdated ->
-                warningMessage $ user <> "/" <> repoName <> ": " <> "Outdated"
-            Repo user repoName FileDoesNotExist ->
-                errorMessage $ user <> "/" <> repoName <> ": " <> "Doesn't have file"
-
-
-checkRepositoriesToUpdate :: Text -> [Text] -> File -> IO [Repo]
-checkRepositoriesToUpdate user repos file =
-    mapM (checkSingleRepository user file) repos
-
-checkSingleRepository :: Text -> File -> Text -> IO Repo
-checkSingleRepository user file repoName = do
-    resp <- fetchFile user repoName $ fileName file
+    resp <- httpBS req
 
     case getResponseStatus resp of
-      Status 200 _ ->
-        if checkSame (fileContents file) $ decodeBody resp then
-          pure $ mkRepo user repoName UpToDate else
-          pure $ mkRepo user repoName Outdated
-      _ -> pure $ mkRepo user repoName FileDoesNotExist
-
-checkSame :: Text -> Text -> Bool
-checkSame orgFile repoFile = orgFile == repoFile
-
-decodeBody :: Response ByteString -> Text
-decodeBody = decodeUtf8 . getResponseBody
+        Status 200 _ -> pure $ Just $ getResponseBody resp
+        _            -> pure Nothing
+
+
+printFileDiff :: UserName -> RepoName -> FileStatus -> IO ()
+printFileDiff userName repoName fileStatus =
+    case fileStatus of
+        UpToDate ->
+            infoMessage $ user <> "/" <> repo <> ": " <> "Up-to-date"
+        Outdated ->
+            warningMessage $ user <> "/" <> repo <> ": " <> "Outdated"
+        FileDoesNotExist ->
+            errorMessage $ user <> "/" <> repo <> ": " <> "Doesn't have file"
+    where
+        user :: Text
+        user = unUserName userName
+
+        repo :: Text
+        repo = unRepoName repoName
+
+
+calculateDiff :: Maybe ByteString -> Maybe ByteString -> FileStatus
+calculateDiff Nothing _ = FileDoesNotExist
+calculateDiff _ Nothing = FileDoesNotExist
+calculateDiff templateFile repoFile =
+    if checkSame templateFile repoFile then
+      UpToDate else
+      Outdated
+
+checkSame :: Maybe ByteString -> Maybe ByteString -> Bool
+checkSame (Just templateFile) (Just repoFile) = templateFile == repoFile
+checkSame _ _                                 = False
```

Comment:
Technically we should never reach this but I like totality. :)

Reply:
@Nimor111 Totality is great :slightly_smiling_face: But I think that this function should take `ByteString` and `Maybe ByteString` instead of two `Maybe`s. If downloading the template file failed, there's no point in working with repos. So you can fail in `sync` early if `templateFile` is `Nothing`, and this should simplify diff calculation.